src/librustdoc/html/render.rs

```diff
@@ -2304,6 +2363,53 @@ fn make_item_keywords(it: &clean::Item) -> String {
     format!("{}, {}", get_basic_keywords(), it.name.as_ref().unwrap())
 }
 
+fn get_index_search_type(item: &clean::Item,
+                         parent: Option<String>) -> Option<IndexItemFunctionType> {
+    match item.inner {
+        clean::FunctionItem(ref f) => Some(get_index_fn_type(&f.decl, None)),
+        clean::MethodItem(ref m) => Some(get_index_fn_type(&m.decl, parent)),
+        clean::TyMethodItem(ref m) => Some(get_index_fn_type(&m.decl, parent)),
+        _ => None
+    }
+}
+
+fn get_index_fn_type(decl: &clean::FnDecl, parent: Option<String>) -> IndexItemFunctionType {
```

Comment:
Yeah this is one of the cooler parts about Rust that I'm a big fan of. Because the expression `return foo` does not ever return, it's allowed to have any type, so you can use it in places where some other type would otherwise be expected. The same goes for expressions like `break` and `continue` as well.